### PR TITLE
Added button to collapse sidebar

### PR DESCRIPTION
**Pull request by kurtisassad**
[Loom walkthrough](https://www.loom.com/share/2bd87e7cd296451e8c2c7907e8a3b374?sid=0f7df67a-3b1f-43ad-880c-50e3ad5ebd46)

## Link to Issue
Closes: #5311

## Description of Changes
- Adds collapsable button for scoped and unscoped pages

## Test Plan
- Go to unscoped page like http://localhost:8080/dashboard/for-you play around with the collapsable button
- Go to scoped page like http://localhost:8080/dydx/discussions play around with collapsable button

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import 'components/sidebar/collapsable_button.scss';
+import React, { useState } from 'react';
+import useSidebarStore from '../../../state/ui/sidebar/index';
+import { CWIconButton } from '../component_kit/cw_icon_button';
+
+export const CollapsableSidebarButtonUnscoped = () => {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:6]: We do not need two separate components if they are pretty much exactly the same. The differences might be handled with props. Please unify CollapsableSidebarButtonScoped and CollapsableSidebarButtonUnscoped by passing type scoped/unscoped

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const { setMenu, menuName, menuVisible, setUserToggledVisibility } =
+    useSidebarStore();
+
+  const [isCollapsed, setIsCollapsed] = useState(false);
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:10]: why do you need additional state if we already have global state to check if sidebar is visible or not. use `menuVisible` instead and get rid of this state

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  function handleToggle() {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:12]: use const instead of function

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    setMenu({ name: menuName, isVisible });
+    setTimeout(() => {
+      setUserToggledVisibility(isVisible ? 'open' : 'closed');
+    }, 200);
+
+    setIsCollapsed(!isCollapsed);
+  }
+
+  return (
+    <div className="hover-box">
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:23]: our convention is that the file, component and first classname, all have the same name. please change it to CollapsableSidebarButtonUnscoped or whatever the file will be named eventually.

File: packages/commonwealth/client/styles/components/sidebar/collapsable_button.scss
Comment (masvelio) [inline packages/commonwealth/client/styles/components/sidebar/collapsable_button.scss:1]: 1. do not put style files into `client/styles...` - we ditched this approach long time ago
2. in `scripts/views/components/sidebar` create `SidebarButton` directory where you put `CollapsableSidebarButtonUnscoped.tsx`, `CollapsableSidebarButtonUnscoped.scss` and `index.ts` that exports `CollapsableSidebarButtonUnscoped.tsx`

File: packages/commonwealth/client/styles/components/sidebar/collapsable_button.scss [hunk 5]
```diff
@@ -0,0 +1,64 @@
+// This is an invisible box that allows us to hover over it in order to make the collapsable button visible
+.hover-box {
```
Comment (masvelio) [inline packages/commonwealth/client/styles/components/sidebar/collapsable_button.scss:2]: we should always encapsulate all styles in the file with parent class, so for example for component CollapsableSidebarButtonUnscoped, the file should be named `CollapsableSidebarButtonUnscoped.scss` and the parent class should be `CollapsableSidebarButtonUnscoped` and then inside every descendant classname. 

```
.CollapsableSidebarButtonUnscoped {
   .other {}
   . classes {}
}
```

This is very important for style encapsulation

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    useSidebarStore();
+
+  const [isCollapsed, setIsCollapsed] = useState(false);
+
+  function handleToggle() {
+    const isVisible = !menuVisible;
+    setMenu({ name: menuName, isVisible });
+    setTimeout(() => {
+      setUserToggledVisibility(isVisible ? 'open' : 'closed');
+    }, 200);
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:17]: why we need this timeout?

File: packages/commonwealth/client/scripts/views/SublayoutHeader.tsx [hunk 7]
```diff
@@ -85,7 +84,7 @@ export const SublayoutHeader = ({ onMobile }: SublayoutHeaderProps) => {
           {isWindowSmallInclusive(window.innerWidth) && (
             <CWDivider isVertical />
           )}
-          {(featureFlags.sidebarToggle || onMobile) && app.activeChainId() && (
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/SublayoutHeader.tsx:88]: shouldn't we remove `featureFlags.sidebarToggle` altogether? from webpack and everywhere

**Review by masvelio — CHANGES_REQUESTED** [review]
1. plus and explore icons are moved down for some reason, they should be on the top of the bar

image

2. because of this overlay I am not able to click anywhere in the sidebar

https://github.com/hicommonwealth/commonwealth/assets/14819225/fb95c686-f946-41a2-8ebe-a6da5cd715d0

3. When you hover over the arrow button we should have pointer cursor like on the other icon buttons.

https://github.com/hicommonwealth/commonwealth/assets/14819225/8756c8f7-3539-4fb0-9346-5a87c253538d


4. On mobile 
  - carret is visible and it is in weird position
  - also I am not able to click ~70% of the screen
  - we have now 2 icons for toggling the sidebar, yours and one in the top navbar. Do we have designs for mobile and how this should work on small devices? 

https://github.com/hicommonwealth/commonwealth/assets/14819225/ef9799e2-2249-4922-9056-73ed7e98b7c6

Summing up: 
1. Code structure has to be fixed
2. Couple bad bugs
3. We need to handle mobile properly

Let me know if you need any guidance here.

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx (hunk 2 above)
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:10]: The issue is that menuVisible seems not to control the sidebar visibility. As the sidebar doesn't react to it but instead always starts off as uncollapsed, and changes to collapsed.

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx (hunk 1 above)
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:6]: Its just a little bit more difficult to follow if we put everything in a single component. This is because there are some funky stuff with the hidden-box that needs to be accounted for in scoped pages.

Anyways, unified the components

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx (hunk 6 above)
Comment (kurtisassad) [reply packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx:17]: Just a style preference. For me it looks a bit jarring if the sidebar collapses too quickly after button press.

Anyways removed it to make the code cleaner

Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButtonUnscoped.tsx]: thanks!

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.scss [hunk 8]
```diff
@@ -0,0 +1,72 @@
+.CollapsableSidebarButtonUnscoped {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.scss:1]: CollapsableSidebarButton

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.scss [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  // This is an invisible box that allows us to hover over it in order to make the collapsable button visible
+  .hover-box {
+    position: fixed;
+    height: 100vh;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.scss:5]: what is expected behaviour? Looking at this line, I would assume that we should hover over the whole edge of the browser and the arrows would appear. However, right now, icon shows up only on icon hover. Not bad for me, but the code does not seem to be doing what it looks like to do

https://github.com/hicommonwealth/commonwealth/assets/14819225/2cef657e-c0a1-48dd-990f-603d9f60ed53

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  const handleToggle = () => {
+    const isVisible = !menuVisible;
+    setMenu({ name: menuName, isVisible });
+    setUserToggledVisibility(isVisible ? 'open' : 'closed');
+  };
+
+  return (
+    <div className="CollapsableSidebarButtonUnscoped">
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx:27]: CollapsableSidebarButton

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+      <div className={divBoxCss}>
+        <CWIconButton
+          iconButtonTheme="black"
+          iconName="caretDoubleLeft"
+          onClick={handleToggle}
+          iconSize="small"
+          className={
+            menuVisible ? `expand-${cssString}` : `collapse-${cssString}`
+          }
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx:36]:
```suggestion
          className={clsx({
            'expand-scoped': menuVisible && isInsideCommunity,
            'expand-unscoped': menuVisible && !isInsideCommunity,
            'collapse-scoped': !menuVisible && isInsideCommunity,
            'collapse-unscoped': !menuVisible && !isInsideCommunity,
          })}
```

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+  const handleToggle = () => {
+    const isVisible = !menuVisible;
+    setMenu({ name: menuName, isVisible });
+    setUserToggledVisibility(isVisible ? 'open' : 'closed');
+  };
+
+  return (
+    <div className="CollapsableSidebarButtonUnscoped">
+      <div className={divBoxCss}>
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx:28]:
```suggestion
      <div
        className={clsx('hover-box', {
          expanded: isInsideCommunity && menuVisible,
        })}
      >
```

Together with css change in `CollapsableSidebarButton.scss`

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  isInsideCommunity: boolean;
+}) => {
+  const { setMenu, menuName, menuVisible, setUserToggledVisibility } =
+    useSidebarStore();
+
+  const cssString = isInsideCommunity ? 'scoped' : 'unscoped';
+  let divBoxCss = 'hover-box';
+  if (isInsideCommunity && menuVisible) {
+    divBoxCss = 'hover-box-expanded';
+  }
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.tsx:18]: remove because of changes below

File: packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.scss [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+
+  .hover-box-expanded {
+    position: fixed;
+    height: 100vh;
+    width: 290px;
+    overflow: hidden;
+    z-index: 5;
+    transition: 0.2s linear;
+    pointer-events: none;
+  }
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/sidebar/CollapsableSidebarButton/CollapsableSidebarButton.scss:21]:
```suggestion
  .hover-box {
    position: fixed;
    height: 100vh;
    width: 57px;
    overflow: hidden;
    z-index: 5;
    transition: 0.2s linear;
    pointer-events: none;

    &.expanded {
      width: 290px;
      transition: 0.2s linear;
    }
  }
```

**Review by masvelio — APPROVED** [review]
1. Thanks for all these changes. I also put some code suggestion that handle css in a bit better and readable way.

2. lets put cursor pointer on arrows on hover
https://github.com/hicommonwealth/commonwealth/assets/14819225/ed9562db-02f2-4455-8085-41e753d0063d

3. For some reason on mobile on some pages (probably unscoped) the old icon is not visible so we are not able to open the left sidebar. Please identify and fix the issue.

https://github.com/hicommonwealth/commonwealth/assets/14819225/4f5da4d0-4f49-4d36-87c3-33bb7f84d97b

Other than that, good job! Approving in advance!

**Comment by mzparacha** [comment]
We have this extra space on mobile, when the collapse button is not shown

image

vs on desktop

image

**Review by mzparacha — COMMENTED** [review]
Found a small issue above, other than that LGTM.

**Review by mzparacha — APPROVED** [review]
